tools/crm_mon_curses.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <stdlib.h>
+#include <crm/crm.h>
+#include <crm/common/curses_internal.h>
+#include <crm/common/output.h>
+#include <glib.h>
+
+#include "crm_mon.h"
+
+#if CURSES_ENABLED
+
+GOptionEntry crm_mon_curses_output_entries[] = {
+    { NULL }
+};
+
+typedef struct curses_list_data_s {
+    unsigned int len;
+    char *singular_noun;
+    char *plural_noun;
+} curses_list_data_t;
+
+typedef struct private_data_s {
+    GQueue *parent_q;
+} private_data_t;
+
+static void
+curses_free_priv(pcmk__output_t *out) {
+    private_data_t *priv = out->priv;
+
+    if (priv == NULL) {
+        return;
+    }
+
+    g_queue_free(priv->parent_q);
+    free(priv);
+}
+
+static bool
+curses_init(pcmk__output_t *out) {
+    private_data_t *priv = NULL;
+
+    /* If curses_init was previously called on this output struct, just return. */
+    if (out->priv != NULL) {
+        return true;
+    } else {
+        out->priv = calloc(1, sizeof(private_data_t));
+        if (out->priv == NULL) {
+            return false;
+        }
+
+        priv = out->priv;
+    }
+
+    priv->parent_q = g_queue_new();
+
+    initscr();
+    cbreak();
+    noecho();
+
+    return true;
+}
+
+static void
+curses_finish(pcmk__output_t *out, crm_exit_t exit_status, bool print, void **copy_dest) {
+    echo();
+    nocbreak();
+    endwin();
+}
+
+static void
+curses_reset(pcmk__output_t *out) {
+    CRM_ASSERT(out->priv != NULL);
+
+    curses_free_priv(out);
+    curses_init(out);
+}
+
+static void
+curses_subprocess_output(pcmk__output_t *out, int exit_status,
+                         const char *proc_stdout, const char *proc_stderr) {
+    if (proc_stdout != NULL) {
+        printw("%s\n", proc_stdout);
+    }
+
+    if (proc_stderr != NULL) {
+        printw("%s\n", proc_stderr);
+    }
+
+    clrtoeol();
+    refresh();
+}
+
+/* curses_version is defined in curses.h, so we can't use that name here */
+static void
+curses_ver(pcmk__output_t *out, bool extended) {
+    if (extended) {
+        printw("Pacemaker %s (Build: %s): %s\n", PACEMAKER_VERSION, BUILD_VERSION, CRM_FEATURES);
+    } else {
+        printw("Pacemaker %s\n", PACEMAKER_VERSION);
+        printw("Written by Andrew Beekhof\n");
+    }
+
+    clrtoeol();
+    refresh();
+}
```

Comment:
--help should already be text everywhere.  For --version, I could just replace the guts of the curses function with one that does plain text, assuming we'll never want a curses-based version function.  For that matter, I could replace all the version functions with text ones if we want.

Reply:
I think if someone explicitly specifies --output-as=html --version, they likely want HTML, or at least we can't be blamed for giving it to them. :) Only console doesn't really make sense with --version.

Reply:
On second thought, we at least need to use curses output for --version, because that happens after `pcmk__output_new` is called and that function calls `curses_init`, which in turn does the curses setup functions.  So if version uses plain text after curses is initialized, it ends up less than ideal.